mms/metric.py

```diff
@@ -124,35 +152,56 @@ def start_recording(self):
 
         # Start recording metrics
         with self.mutex:
-            if self.write_to == 'csv':
-                filename = os.path.join('metrics', 'mms_' + self.name + '.csv')
-                if not os.path.exists(os.path.dirname(filename)):
-                     os.makedirs(os.path.dirname(filename))
-                with open(filename, 'a') as csvfile:
-                    csvwriter = csv.writer(csvfile, delimiter=',')
-                    csvwriter.writerow([utcnow, metric])
-            elif self.write_to == 'cloudwatch':
-                logger.info('Metric %s for last %s seconds is %f, writing to AWS CloudWatch...' %
-                    (self.name, self.interval_sec, metric))
-                try:
-                    self.client.put_metric_data(
-                        Namespace=self.namespace,
-                        MetricData=[
-                            {
-                                'MetricName': self.name,
-                                'Timestamp': utcnow,
-                                'Value': metric,
-                                'Unit': 'Count'
+            # metric starts being reported
+            if self.interval_datapoints_count != 0:
+                if self.write_to == 'csv':
+                    filename = os.path.join('metrics', 'mms_' + self.name + '.csv')
+                    if not os.path.exists(os.path.dirname(filename)):
+                         os.makedirs(os.path.dirname(filename))
+                    with open(filename, 'a') as csvfile:
+                        csvwriter = csv.writer(csvfile, delimiter=',')
+                        csvwriter.writerow([utcnow, metric])
+                elif self.write_to == 'cloudwatch':
+                    logger.info('Metric %s for last %s seconds is %f, writing to AWS CloudWatch...' %
+                        (self.name, self.interval_sec, metric))
+                    try:
+                        metric_data = {
+                            'MetricName': self.name,
+                            'Timestamp': utcnow,
+                            'Unit': self.unit,
+                            'StatisticValues': {
+                                'SampleCount': self.interval_datapoints_count,
+                                'Sum': self.interval_metric_aggregate,
```

Comment:
- how does this work for % values (i think it won't)
- we should report disk/mem used/free mb values as single data points not stats
- it looks like we will not be able to get percentiles this way, and would have to calculate and report those as a separate group of metrics.  i'm ok with deferring those to future enhancement.

Reply:
1. It will work for % values. The only percentage value we are collecting is cpu/memory/disk usage which is reported by python library. It is just a value of float number smaller 1, there should be no difference.
2. Changed.
3. Yea, we can do it later considering current timeline.